### PR TITLE
Add persistence strategy to KeyValue Storage.

### PR DESCRIPTION
**Pull request by wangff**

### CODE REVIEW

**Comment by wangff** [comment]
@hugagaga this pull request is ready for review. Thanks!

File: src/KeyValueStore/persistence.h [hunk 1]
```diff
@@ -0,0 +1,22 @@
+#ifndef CSCI499_FEI_SRC_KEYVALUESTORE_PERSISTENCE_H_
+#define CSCI499_FEI_SRC_KEYVALUESTORE_PERSISTENCE_H_
+
+#include "persistence_abstraction.h"
+
+namespace cs499_fei {
+// Delimiter to separate the key and value
+const char kDelimKeyValue = '#';
```
Comment (hugagaga) [inline src/KeyValueStore/persistence.h:8]: Remove unused constant.

File: src/KeyValueStore/keyvaluestore_server.cc [hunk 2]
```diff
@@ -55,14 +84,22 @@ void RunServer() {
 
   LOG(INFO) << "Server listening on " << server_address;
 
+  // register signal SIGINT and signal handler
+  signal(SIGINT, signalHandler);
+  shutdownHandler = [&](int signal) {
+    std::cout << "Server shutdown... " <<std::endl;
+    service.store();
```
Comment (hugagaga) [inline src/KeyValueStore/keyvaluestore_server.cc:91]: What if multiple clients store data to the same file at the same time?  This might be a rare case.

Comment (wangff) [reply src/KeyValueStore/keyvaluestore_server.cc:91]: You are right, this is a case. However, this signalHandler called in the kvstore_server, which is just run once and has nothing to do with clients. When kvstore_server is running, all the data is running in memory, and all clients interacted with kvstore_server and in-memory data. When kvstore_server is terminated, signalHanlder will store the data into the local file, and no clients could connect with the server and no clients could access the data. Thus, in another word, multiple clients just access the in-memory kv_store map. When kvstore_store is terminated, all data will be written to the file.

File: src/KeyValueStore/persistence.h (hunk 1 above)
Comment (wangff) [reply src/KeyValueStore/persistence.h:8]: It's done.

**Pull request merged**